### PR TITLE
Compile README.md via markdown-unlit 

### PR DESCRIPTION
**Pull request by eborden**
Keeping a `README.md` up to date with your library is an annoying
endeavor. They are constantly falling out of sync. Luckily we have type
checking to ammend this for us. By utilizing `markdown-unlit` we can
compile the `README.md` for this project and keep it in sync with
library updates. This keeps our examples "working" and ensures less
dissonance for users.

This toolchain involves actually storing `README.md` details in
`README.lhs` and then symlinking `README.md` to that file. `github` and
other tools pick up on the symlink and transparently display the
appropriate markdown.

### CODE REVIEW

**Comment by eborden** [comment]
Looks like `weeder` and `hlint` are having issue with `lhs`.

**Comment by eborden** [comment]
grumble grumble
```
== Section test:readme ==
Redundant build-depends entry
* markdown-unlit
```

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,8 @@ test:
 
 .PHONY: lint
 lint:
-	hlint .
+	hlint library/.
+	hlint tests/.
```
Comment (pbrisbin) [inline Makefile]: `hlint library tests` doesn't work?

**Comment by eborden** [comment]
I continue to love that every @ndmitchell tool spits out a default `yaml` file with the current errors ignored. Such a fantastic onboarding experience.

**Comment by ndmitchell** [comment]
Glad the default is helping! Although both HLint and Weeder are expected to work with lhs, so please do report any bugs - it's certainly a less tested path but both have at least one test.

**Pull request merged**